ports/unix/coverage.c

```diff
@@ -538,13 +538,32 @@ static mp_obj_t extra_coverage(void) {
     {
         mp_printf(&mp_plat_print, "# binary\n");
 
-        // call function with float and double typecodes
+        // call function with various typecodes
         float far[1];
         double dar[1];
+        int8_t i8ar[1];
+        uint8_t u8ar[1];
+        int16_t i16ar[1];
+        uint16_t u16ar[1];
+        int32_t i32ar[1];
+        uint32_t u32ar[1];
         mp_binary_set_val_array_from_int('f', far, 0, 123);
-        mp_printf(&mp_plat_print, "%.0f\n", (double)far[0]);
         mp_binary_set_val_array_from_int('d', dar, 0, 456);
+        mp_binary_set_val_array_from_int('b', i8ar, 0, -1);
```

Comment:
This function is only called from stm32/adc.c so an alternative fix for this would be to delete it from binary.h and binary.c and make it a static function in adc.c.